specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/mfe.json

```diff
@@ -0,0 +1,1719 @@
+            "type": "object"
+          }
+        },
+        "errorResponse": {
+          "$ref": "#/definitions/ErrorResponse"
+        }
+      }
+    },
+    "RootError": {
+      "type": "object",
+      "properties": {
+        "code": {
+          "type": "string"
+        },
+        "severity": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "message": {
+          "type": "string"
+        },
+        "messageFormat": {
+          "type": "string"
+        },
+        "messageParameters": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "referenceCode": {
+          "type": "string"
+        },
+        "detailsUri": {
+          "format": "uri",
+          "type": "string"
+        },
+        "target": {
+          "type": "string"
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RootError"
+          }
+        },
+        "innerError": {
+          "$ref": "#/definitions/InnerErrorResponse"
+        }
+      }
+    },
+    "ErrorResponse": {
+      "type": "object",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/RootError"
+        },
+        "correlation": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "environment": {
+          "type": "string"
+        },
+        "location": {
+          "type": "string"
+        },
+        "time": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "componentName": {
+          "type": "string"
+        }
+      }
+    },
+    "ContentsType": {
+      "enum": [
+        "AzureBlob",
+        "AzureDataLake",
+        "AzureDataLakeGen2",
+        "AzureFile",
+        "AzureMySql",
+        "AzurePostgreSql",
+        "AzureSqlDatabase",
+        "GlusterFs"
+      ],
+      "type": "string"
+    },
+    "CredentialsType": {
+      "enum": [
+        "AccountKey",
+        "Certificate",
+        "Sas",
+        "ServicePrincipal",
+        "SqlAdmin"
+      ],
+      "type": "string"
+    },
+    "AccountKeySection": {
+      "type": "object",
+      "properties": {
+        "key": {
+          "type": "string",
+          "x-ms-secret": true
+        }
+      }
+    },
+    "CertificateSection": {
+      "required": [
+        "clientId",
+        "tenantId",
+        "thumbprint"
+      ],
+      "type": "object",
+      "properties": {
+        "authorityUrl": {
+          "type": "string"
+        },
+        "resourceUri": {
+          "type": "string"
+        },
+        "tenantId": {
+          "format": "uuid",
+          "type": "string"
+        },
+        "clientId": {
+          "format": "uuid",
+          "type": "string"
+        },
+        "certificate": {
+          "type": "string",
+          "x-ms-secret": true
```

Comment:
secrets should also be annoted with the appropriate x-ms-mutability to indicate they are not returned on responses or GET calls.  See https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-mutability